pydantic/validators.py

```diff
@@ -523,6 +530,44 @@ def pattern_validator(v: Any) -> Pattern[str]:
         raise errors.PatternError()
 
 
+NamedTupleT = TypeVar('NamedTupleT', bound=NamedTuple)
+
+
+def make_named_tuple_validator(type_: Type[NamedTupleT]) -> Callable[[Tuple[Any, ...]], NamedTupleT]:
+    from .main import create_model
+
+    # A named tuple can be created with `typing,NamedTuple` with types
+    # but also with `collections.namedtuple` with just the fields
+    # in which case we consider the type to be `Any`
+    named_tuple_annotations: Dict[str, Type[Any]] = getattr(type_, '__annotations__', {k: Any for k in type_._fields})
+    field_definitions: Dict[str, Any] = {
+        field_name: (field_type, ...) for field_name, field_type in named_tuple_annotations.items()
+    }
+    NamedTupleModel: Type['BaseModel'] = create_model('NamedTupleModel', **field_definitions)
+
+    def named_tuple_validator(values: Tuple[Any, ...]) -> NamedTupleT:
+        dict_values: Dict[str, Any] = dict(zip(named_tuple_annotations, values))
+        validated_dict_values: Dict[str, Any] = dict(NamedTupleModel(**dict_values))
+        return type_(**validated_dict_values)
+
+    return named_tuple_validator
+
+
+def make_typed_dict_validator(type_: Type['TypedDict']) -> Callable[[Any], Dict[str, Any]]:
+    from .main import create_model
+
+    field_definitions: Dict[str, Any] = {
+        field_name: (field_type, ... if type_.__total__ else None)
```

Comment:
for performance and clarity
```suggestion
    default_value = ... if type_.__total__ else None
    field_definitions: Dict[str, Any] = {
        field_name: (field_type, default_value)
```